source/common/http/conn_manager_impl.cc

```diff
@@ -335,13 +335,11 @@ ConnectionManagerImpl::ActiveStream::~ActiveStream() {
     log_handler->log(request_headers_.get(), response_headers_.get(), request_info_);
   }
 
-  if (active_span_) {
-    if (request_info_.healthCheck()) {
-      connection_manager_.config_.tracingStats().health_check_.inc();
-    } else {
-      Tracing::HttpTracerUtility::finalizeSpan(*active_span_, *request_headers_, request_info_,
-                                               *this);
-    }
+  if (request_info_.healthCheck()) {
+    connection_manager_.config_.tracingStats().health_check_.inc();
+  } else {
+    Tracing::HttpConnManFinalizerImpl finalizer{*request_headers_, request_info_, *this};
```

Comment:
If this is the only place we do this we could consider just keeping track of the tracing decision from where `active_span_` was set, but then you're essentially going back to treating this as nullptr. In general though it seems that there is the pattern of "some setup work; do something on a span that might be null;", so it would be useful to have a cheap way to know if actively tracing. `iSNull` seems reasonable in the absence of some other object with trace context info.